modules/gitops-configuring-sscsi-driver-to-mount-secrets-from-aws-secrets-manager.adoc

```diff
@@ -0,0 +1,186 @@
+----
+
+.Example output
+[source,terminal]
+----
+us-west-2
+----
+====
+
+.. Retrieve the OIDC provider by running the following command:
++
+[source,terminal]
+----
+$ oc get --raw=/.well-known/openid-configuration | jq -r '.issuer'
+----
++
+.Example output
+[source,terminal]
+----
+https://<oidc_provider_name>
+----
+Copy the OIDC provider name `<oidc_provider_name>` from the output to use in the next step.
+
+.. Use the `ccoctl` tool to process the credentials request by running the following command:
++
+[source,terminal]
+----
+$ ccoctl aws create-iam-roles \
+    --name my-role --region=<aws_region> \
+    --credentials-requests-dir=credentialsrequest-dir-aws \
+    --identity-provider-arn arn:aws:iam::<aws_account>:oidc-provider/<oidc_provider_name> --output-dir=credrequests-ccoctl-output
+----
++
+.Example output
+[source,terminal]
+----
+2023/05/15 18:10:34 Role arn:aws:iam::<aws_account_id>:role/my-role-my-namespace-aws-creds created
+2023/05/15 18:10:34 Saved credentials configuration to: credrequests-ccoctl-output/manifests/my-namespace-aws-creds-credentials.yaml
+2023/05/15 18:10:35 Updated Role policy for Role my-role-my-namespace-aws-creds
+----
++
+Copy the `<aws_role_arn>` from the output to use in the next step. For example, `arn:aws:iam::<aws_account_id>:role/my-role-my-namespace-aws-creds`.
+
+.. Check the role policy on AWS to confirm the <aws_region> of `"Resource"` in the role policy matches the cluster region:
++
+.Example
+[source,json]
+----
+{
+	"Version": "2012-10-17",
+	"Statement": [
+		{
+			"Effect": "Allow",
+			"Action": [
+				"secretsmanager:GetSecretValue",
+				"secretsmanager:DescribeSecret"
+			],
+			"Resource": "arn:aws:secretsmanager:<aws_region>:<aws_account_id>:secret:your-secret-xxxxxx"
+		}
+	]
+}
+----
+
+.. Bind the service account with the role ARN by running the following command:
++
+[source,terminal]
+----
+$ oc annotate -n my-namespace sa/aws-provider eks.amazonaws.com/role-arn="<aws_role_arn>"
+----
++
+.Example
+[source,terminal]
+----
+$ oc annotate -n dev sa/default eks.amazonaws.com/role-arn="<aws_role_arn>"
+----
++
+.Example output
+[source,terminal]
+----
+serviceaccount/default annotated
+----
+
+. Create a namespace-scoped `SecretProviderClass` resource to define your secrets store provider:
+
+.. Create a `secret-provider-class-aws.yaml` file in the same directory where the target deployment is located in your GitOps repository. For example, consider that you create a `SecretProviderClass` resource in `/environments/dev/apps/app-taxi/services/taxi/base/config` directory of your GitOps repository:
++
+.Example `secret-provider-class-aws.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-aws-provider # <1>
+  namespace: dev        # <2>
+spec:
+  provider: aws         # <3>
+  parameters:           # <4>
+    objects: |
+      - objectName: "gitops-secret" # <5>
```

Comment:
In the previous step, you mentioned the secret as `your-secret-xxxxx` and here you used a different name which is `gitops-secret`. We should keep them in same names for same thing in same context. If you have created a secret named `gitops-secret`, the previous step you should mention it like `gitops-secret-xxxxx`. If you want to use a more universal/meaningful name, you could put `your-secret-xxxx` and `-objectName: "your-secret"`, so that people won't get confused by the different values.